### PR TITLE
Added regression tests for hydrogenbondautocorrel

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
Ready to merge or do you want to add more? +0.9% is good!

File: testsuite/MDAnalysisTests/analysis/test_hydrogenbondautocorrel.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+import numpy as np
+
+import MDAnalysis as mda
+from MDAnalysis.analysis.hbonds import HydrogenBondAutoCorrel as HBAC
+
+
+class TestHydrogenBondAutocorrel(object):
+    def setUp(self):
+        u = self.u = mda.Universe(TRZ_psf, TRZ)
+        self.H = u.atoms.Hn
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/analysis/test_hydrogenbondautocorrel.py]: I wouldn't rely on the quick selectors and rather do explicit `select_atoms`.

File: testsuite/MDAnalysisTests/analysis/test_hydrogenbondautocorrel.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                      sample_time=0.06,
+        )
+
+    def test_bond_type_VE(self):
+        assert_raises(ValueError,
+                      HBAC, self.u,
+                      hydrogens=self.H,
+                      acceptors=self.O,
+                      donors=self.N,
+                      bond_type='marzipan',
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/analysis/test_hydrogenbondautocorrel.py]: haha

**Comment by orbeckst** [comment]
For some reason, [intermittent bonds](https://coveralls.io/builds/7202743/source?filename=miniconda%2Fenvs%2Fpyenv%2Flib%2Fpython2.7%2Fsite-packages%2FMDAnalysis%2Fanalysis%2Fhbonds%2Fhbond_autocorrel.py#L369) are not tested, even though you have tests.

**Comment by richardjgowers** [comment]
Ok this can merge once it passes

**Comment by orbeckst** [comment]
One test failed. Possibly decrease requested comparison accuracy.

**Comment by richardjgowers** [comment]
Ok I redid the test (and was less lazy about it), should pass now.

**Pull request merged**